packages/kbn-check-mappings-update-cli/current_mappings.json

```diff
@@ -2345,6 +2345,21 @@
       }
     }
   },
+  "cloud-security-posture-settings": {
+    "dynamic": false,
+    "properties": {
+      "rules_states": {
+        "properties": {
+          "key": {
+            "type": "keyword"
+          },
+          "muted": {
+            "type": "boolean"
+          }
+        }
+      }
+    }
+  },
```

Comment:
I haven't seen `"key": { "type": "keyword" }` used likes this before. Does this enable search over document property keys as though they are values?


From what I can tell your document will look like:

```
{
  <rule-id?>: { "muted": <boolean-value> }
 }
```

Which does not match the above structure. I have a couple of questions

1. Do want to search over both rule ID(?) and muted values?
2. If you want to search over both, I think we need to restructure your SO document to match your mappings. Perhaps something like:

```
{
  "ruleId": { "type": "keyword" },
  "muted": { "type": "boolean" }
}
```

Let me know if I'm missing something though!

Reply:
Thanks @jloleysens, you are right. The mapping is incorrect. My goal is to define a mapping for a key that is generated dynamically.Thanks @jloleysens, you are right. The mapping is incorrect. My goal is to define a mapping for a key that is generated dynamically.
As you can see in the example below `cis_k8s;v3.0.0;9.1` and `cis_aws;v3.0.0;0.1` are defined dynamically. 
```
{
  id: "csp-internal-settings",
  type: "cloud-security-posture-settings",
  attributes: {
    rules_states: {
      "cis_aws;v3.0.0;0.1": {
        muted: true,
      },
      "cis_k8s;v3.0.0;9.1: {
        muted: true,
      }
    },
  },
}
```

Reply:
>Do want to search over both rule ID(?) and muted values?
If you want to search over both, I think we need to restructure your SO document to match your mappings. Perhaps something like:

I prefer to use the rule ids as keys since I prefer [extract them easily](https://github.com/elastic/kibana/pull/172144/files#diff-e9a19f070132f113562d778a0c97840651e8af13306d06367618378d302fdfb2R47). 

If it's not possible to create a mapping that facilitates searching by the key values, how would you propose constructing the mapping while retaining the current object structure?